pkg/kubelet/kubelet.go

```diff
@@ -461,7 +461,10 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 			clusterDNS = append(clusterDNS, ip)
 		}
 	}
-	httpClient := &http.Client{}
```

Comment:
while we don't have have the information to do TLS verification for container lifecycle requests, constructing an insecure getter here and passing it to `NewKubeGenericRuntimeManager` increases risk that it will be used for purposes other than getting container lifecycle http endpoints

I would either rename this to something like insecureContainerLifecycleHttpClient (all the way through the stack), and/or move the construction much closer to the point of use